### PR TITLE
Fix errors for some uncommon variants

### PR DESCRIPTION
**Pull request by federkasten**
This PR fixes errors in translation of some uncommon variants;
variants which cross over CDS boundary, insertion variants which evoke nonsense mutation, and synonymous mutations.

I have also added test cases for these variants.

Though it should take more elegant way to fix these problems, my fixes enable to translate them well.

Thanks.

### CODE REVIEW

File: test/varity/vcf_to_hgvs_test.clj [hunk 1]
```diff
@@ -33,15 +46,16 @@
...
         "chr1" 42752620 "T" "TGGAGTTC" '("NM_001146289:c.1383_1389dupGAACTCC"
                                          "NM_001243246:c.1383_1389dupGAACTCC"
                                          "NM_022356:c.1383_1389dupGAACTCC") ; cf. rs137853953 (-)
 
         ;; Insertion
         "chr1" 69567 "A" "AT" '("NM_001005484:c.477_478insT")
         "chr3" 122740443 "G" "GAGA" '("NM_024610:c.1368_1369insTCT"
-                                      "NM_001320728:c.1284_1285insTCT") ;cf. rs16338 (-)
+                                      "NM_001320728:c.1284_1285insTCT") ; cf. rs16338 (-)
+        "chr7" 152247986 "G" "GT" '("NM_170606:c.2447dupA") ; cf. rs150073007 (-)
```
Comment (totakke) [inline test/varity/vcf_to_hgvs_test.clj]: Please move this line to "duplication" section, L53.

File: test/varity/vcf_to_hgvs_test.clj [hunk 2]
```diff
@@ -91,12 +105,16 @@
...
+        "chr12" 25245350 "C" "A" '("p.G12V") ; cf. rs121913529 (-)
+        "chr17" 7676147 "G" "A" '("p.A74="
+                                  "p.A35=") ; cf. rs786201577 (synonymous)
 
         ;; deletion
         "chr1" 1286041 "CCTT" "C" '("p.F227del")
         "chr3" 198153259 "GGCAGCAGCA" "G" '("p.Q82_Q84del"); cf. rs56683636 (+)
         "chr1" 247815239 "AAGG" "A" '("p.S163del") ; cf. rs35979231 (-)
         "chr1" 84574315 "CGCAGCGCCA" "C" '("p.L31_L33del"); cf. rs3217269 (-)
+        "chr6" 33086236 "TA" "T" '("p.*259=") ; cf. rs67523850 (deletion in border of UTR)
```
Comment (totakke) [inline test/varity/vcf_to_hgvs_test.clj]: Please move this line to "substitution" section, L111.

File: test/varity/vcf_to_hgvs_test.clj [hunk 3]
```diff
@@ -106,6 +124,7 @@
         "chr3" 73062352 "T" "TTGG" '("p.L91_E92insV") ; cf. rs143235716 (+)
         "chr3" 122740443 "G" "GAGA" '("p.P456_Q457insS"
                                       "p.P428_Q429insS") ; cf. rs71270423 (-)
+        "chr7" 152247986 "G" "GT" '("p.Y816*fs*?") ; cf. rs150073007 (-) ; this shold be nonsense mutation "p.Y816*"?
```
Comment (totakke) [inline test/varity/vcf_to_hgvs_test.clj]: Please move this line to "frame shift" section, L140.

File: src/varity/vcf_to_hgvs/protein.clj [hunk 4]
```diff
@@ -231,13 +253,14 @@
...
-        ter-site (-> (:alt-prot-seq seq-info)
+        ter-site (-> seq-info
+                     format-alt-prot-seq
                      (subs (dec (+ ppos offset)))
                      (string/index-of "*"))]
     (mut/protein-frame-shift (mut/->long-amino-acid ref)
                              (coord/protein-coordinate (+ ppos offset))
                              (mut/->long-amino-acid alt)
-                             (if ter-site
+                             (if (and ter-site (< 1 ter-site))
```
Comment (totakke) [inline src/varity/vcf_to_hgvs/protein.clj]: I think `(pos? ter-site)` is correct. `ter-site` here is zero-based number.

Comment (federkasten) [inline src/varity/vcf_to_hgvs/protein.clj]: According to hgvs.org, `fs*1` should be described as a nonsense mutation.
(e.g. `E746*`)
Current implementation translates such variants to frame shift mutations.
(e.g. `E746*fs*?`)

We should fix this translation issue eventually. L139 in `vcf_to_hgvs_test.clj` is a test case for this.

cf .http://varnomen.hgvs.org/recommendations/protein/variant/frameshift/

> the description of a frame shift variant can not contain “fsTer1”, such a variant is a nonsense variant (see Substitution). The shortest frame shift variant possible contains “fsTer2”

Comment (totakke) [inline src/varity/vcf_to_hgvs/protein.clj]: You are right, and I understand it. `fs*1` is not frame-shift, but `(< 1 ter-site)` here also excludes `fs*2`.

Comment (totakke) [inline src/varity/vcf_to_hgvs/protein.clj]: You are right, and I understand it. `fs*1` is not frame-shift, but `(< 1 ter-site)` here also excludes `fs*2`.

Comment (federkasten) [inline src/varity/vcf_to_hgvs/protein.clj]: Sorry. I have a misunderstanding. I will fix it soon. Thanks.

**Comment by totakke** [comment]
Thanks!

**Pull request merged**